pkg/operator/operator.go

```diff
@@ -143,18 +148,31 @@ func New(
 	eventBroadcaster.StartLogging(glog.Infof)
 	eventBroadcaster.StartRecordingToSink(&coreclientsetv1.EventSinkImpl{Interface: kubeClient.CoreV1().Events("")})
 
+	newRecorder := func(kind string) events.Recorder {
+		return events.NewRecorder(kubeClient.CoreV1().Events("openshift-machine-config-operator"), "machine-config-operator", &corev1.ObjectReference{
+			Kind:       kind,
+			Name:       "machine-config-operator",
+			Namespace:  "openshift-machine-config-operator",
+			APIVersion: "apps/v1",
```

Comment:
This is wrong, you're hardcoding `APIVersion` and you're using `newRecorder` for kinds outside `apps/v1` group. As mentioned elsewhere, I'd suggest creating a single recorder tied with just the MCO deployment, which should be sufficient.